core/client/src/main/java/alluxio/client/lineage/LineageFileOutStream.java

```diff
@@ -42,7 +44,21 @@
    * @throws IOException if an I/O error occurs
    */
   public LineageFileOutStream(AlluxioURI path, OutStreamOptions options) throws IOException {
-    super(path, updateOutStreamOptions(options));
+    this(FileSystemContext.INSTANCE, path, options);
+  }
+
+  /**
+   * Creates a new file output stream when lineage is enabled.
+   *
+   * @param context file system context
+   * @param path the path of the file
+   * @param options the set of options specific to this operation
+   * @throws IOException if an I/O error occurs
+   */
+  public LineageFileOutStream(FileSystemContext context, AlluxioURI path, OutStreamOptions options)
```

Comment:
Yes, they will be first. The codebase is currently in an inconsistent state (I was putting them last by convention until Jiri pointed out that the normal convention is to put them first). I plan on fixing it once all the DI PRs are merged